cmd/log.go

```diff
@@ -111,8 +75,13 @@ func Log(c *git.Client, args []string) error {
 		opts.MaxCount = &mc
 	}
 
+	printCommit, err := git.GetCommitPrinter(c, format)
```

Comment:
maybe instead we should have a `Commit.Format(c, format) (string, error)` function and use that inside the unexported printCommit? `GetCommitPrinter` feels like an internal implementation thing that shouldn't be exported, while `Format` feels like it belongs in on the object.

Reply:
It's not possible to convert all of the pre-built formats, such as medium, into pretty print formats, so we can't get away with just a single Commit.Format() method. I'm thinking that if we can have a handful of methods that this will be ok: Commit.[Medium|Full|Fullest]Format(), Commit.Format()

Reply:
Okay, that seems reasonable.. they don't need to be exported, though.  Format can just delegate to the other functions if passed one of the special keywords. 

Out of curiosity, what's the reason they can't be?

Reply:
Medium, for example has the "Merges:" part that can't be described using the pretty format tags. That isn't necessarily an issue as it can be injected by the command. The spec defines that the Merges: section should come immediately before "Author:" This is a hack, but doable.

The problem comes with the indenting of the commit messages. There's no apparent way with a format string to specify an indent for each line of the output that is being interpolated with the tag.